File: librarymanager/librarymanager-be/src/main/java/ru/elenacherev/librarymanager/api/LibrarianApi.java
package ru.elenacherev.librarymanager.api;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import ru.elenacherev.librarymanager.api.dto.Librarian;

public interface LibrarianApi {

	Librarian getLibrarian(
			@NotNull Long librarianId
	);

	Page<Librarian> searchLibrarians(
			Pageable pageable
	);

	Librarian createLibrarian(
			@Valid Librarian librarian
	);

	Librarian saveLibrarian(
			@NotNull Long librarianId, 
			@Valid Librarian librarian
	);

}

File: librarymanager-api/src/main/java/ru/elenacherev/librarymanager/api/LibrarianApi.kt
package ru.elenacherev.librarymanager.api

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import ru.elenacherev.librarymanager.api.dto.Librarian
import java.util.*
import javax.validation.Valid

interface LibrarianApi {

    fun getLibrarian(
        librarianId: UUID
    ): Librarian?

    fun searchLibrarians(
        pageable: Pageable
    ): Page<Librarian>

    fun createLibrarian(
        librarian: @Valid Librarian
    ): Librarian

    fun saveLibrarian(
        librarianId: UUID,
        librarian: @Valid Librarian
    ): Librarian
}